### ISSUE TITLE
[DocDB] [YSQL] org.yb.pgsql.TestPgRegressTablegroup fails in multiple build types, cannot decode value type kGroupEnd

### ISSUE THREAD

**Issue by mbautin** [yugabyte/yugabyte-db#14862]
Jira Link: [DB-4155](https://yugabyte.atlassian.net/browse/DB-4155)
### Description

x86_64 macOS release:
https://gist.githubusercontent.com/mbautin/190048c188690a44da31fa43ab8cf0c2/raw
pgregress part only: https://gist.githubusercontent.com/mbautin/172bf6d956705c47c9a75b8676a082dc/raw
Important part: `Query error: Cannot decode value type kGroupEnd from the key encoding format: "!"`

---

x86_64 clang15 almalinux8 release LTO with linuxbrew:
https://gist.githubusercontent.com/mbautin/52cd0689c58ae5578ce2756379f1dce0/raw
pgregress part only: https://gist.githubusercontent.com/mbautin/0da2142c6c6c23ab3c356b74a1aca19f/raw
Important part: `Query error: Cannot decode value type kGroupEnd from the key encoding format: "!"`
 
--

aarch64 clang15 almalinux8 debug:
https://gist.githubusercontent.com/mbautin/7593645e6fd1ef9b88c48e692f07424f/raw
pgregress part only: https://gist.githubusercontent.com/mbautin/acdb5495c570123a79bbb04be170f092/raw
Important part: `Timed out: Timed out waiting for Create Table`

--
aarch64 clang15 almalinux8 release:
https://gist.githubusercontent.com/mbautin/e229fa0b3c2d5b33b8b67f31fc7ab54e/raw
pgregress part only: https://gist.githubusercontent.com/mbautin/4699b5275ed738ad1d07c6e14eebe6de/raw
Important part: `Query error: Cannot decode value type kGroupEnd from the key encoding format: "!"`

**Comment by mbautin** [comment]
Detailed stack trace when running clang15 release non-linuxbrew x86_64 almalinux build with libbacktrace turned on on commit 0546ca8c02d2847f5c06c28ca8c0f5bde14b729a with the following patch applied:
```
diff --git a/src/yb/docdb/primitive_value.cc b/src/yb/docdb/primitive_value.cc
index f111b487d0..04ffde01d5 100644
--- a/src/yb/docdb/primitive_value.cc
+++ b/src/yb/docdb/primitive_value.cc
@@ -1083,11 +1083,15 @@ Status KeyEntryValue::DecodeKey(Slice* slice, KeyEntryValue* out) {

     IGNORE_SPECIAL_KEY_ENTRY_TYPES;
   }
-  return STATUS_FORMAT(
+
+  // DEBUG mbautin
+  auto s = STATUS_FORMAT(
       Corruption,
       "Cannot decode value type $0 from the key encoding format: $1",
       type,
       ToShortDebugStr(input_slice));
+  CHECK_OK(s);
+  return s;
 }
```
Stack trace:
https://gist.githubusercontent.com/mbautin/567b7e9da87e85f052f7e585a6885cb6/raw

**Comment by mbautin** [comment]
Command to re-run the test:
```
./yb_build.sh release --clang15 --no-linuxbrew  --java-test TestPgRegressTablegroup#testPgRegressTablegroup -n 1000 --tp=4 YB_EXTRA_DAEMON_FLAGS="--use_libbacktrace"
```

**Comment by mbautin** [comment]
Probably the same root cause as #12899.

**Comment by jasonyb** [comment]
Resolved by recent commit 3d9d8bae563b13010831a5c45ce44b4070126eca.
